src/metaxy/graph/diff/diff_models.py

```diff
@@ -201,7 +201,7 @@ def to_struct(self) -> dict[str, Any]:
                     {
                         "key": field.field_key.to_string(),
                         "version": field.old_version or "",
-                        "code_version": field.old_code_version or 0,
+                        "code_version": field.old_code_version or "",
```

Comment:
This is definitely wrong, we should never have an empty code version. 

It should default to a known shared constant like "0". 

More importantly, **we should not randomly assume the default code version around the codebase** in places **where it should already be set at this point**. 

If a dictionary that **should have it** doesn't actually have this field, we should just fail naturally, because this means something went really wrong in the previous code paths that should have been validating this invariant.